metricflow/dataset/dataset_classes.py

```diff
@@ -34,6 +34,24 @@ def metric_time_dimension_instances(self) -> Sequence[TimeDimensionInstance]:
             if time_dimension_instance.spec.element_name == DataSet.metric_time_dimension_name()
         )
 
+    @property
+    def metric_time_instance_for_time_constraint(self) -> TimeDimensionInstance:
+        """Returns the metric time instance that should be used for a time constraint if one was requested."""
+        metric_time_instances = sorted(
+            (
+                instance
+                for instance in self.metric_time_dimension_instances
+                if not instance.spec.has_custom_grain and not instance.spec.date_part
+            ),
+            key=lambda x: x.spec.base_granularity_sort_key,
+        )
+
+        assert (
```

Comment:
It seems a little odd to have a property that can raise an exception without having a way to know if it's safe to access the property. How about returning an optional value?